Add RuboCop performance

tagliala opened the pull request.

tagliala committed:
Fix safe offenses in non-production code

- Performance/RegexpMatch
- Performance/StringIdentifierArgument

tagliala committed:
Fix safe offenses in production code

- Performance/BlockGivenWithExplicitBlock
- Performance/RegexpMatch
- Performance/ReverseEach
- Performance/StringReplacement

tagliala committed:
Fix unsafe offenses in non-production code

- Performance/StringInclude

tagliala committed:
Fix unsafe offenses in production code

- Performance/RedundantMerge

tagliala committed:
Add RuboCop performance

tagliala commented on gemfiles/rails_70/Gemfile.lock:
```diff
@@ -219,6 +197,7 @@ GEM
 PLATFORMS
   arm64-darwin-22
   java
+  ruby
```
Looks like an unintended change, but I can see ruby in arbre:

https://github.com/activeadmin/arbre/blob/181e9ebd4d4a12ae3db7a2f27dc4542d03586ef9/Gemfile.lock#L294

Let me know if this is not a problem

javierjulio replied:
I'm not sure what causes it. I'm sure it's fine as is. If it's a problem we can address the removal separately. Thanks!

tagliala replied:
This has been added because I have a setting to prefer ruby builds always on my bundler configuration

```
BUNDLE_FORCE_RUBY_PLATFORM: "true"
```

but should not be a problem

tagliala commented on lib/inherited_resources/class_methods.rb:
```diff
@@ -229,23 +229,23 @@ def belongs_to(*symbols, &block)
       def singleton_belongs_to(*symbols, &block)
         options = symbols.extract_options!
-        options.merge!(singleton: true)
+        options[:singleton] = true
```
This line is already missing coverage

javierjulio replied:
This has coverage now by adding a class for this feature but has no explicit test.

tagliala commented on test/url_helpers_test.rb:
```diff
@@ -11,6 +11,12 @@ class UniversesController < InheritedResources::Base
+class GalaxiesController < InheritedResources::Base
+  singleton_belongs_to :universe
+end
+
```
this is going to cover the missing line, even if there are not explicit tests, but I do not understand if this makes sense. Not familiar with "singleton belongs_to"

I guess this test needs a `test_url_helpers_on_singleton_belongs_to` but some guidance could help

javierjulio replied:
Neither am I. I haven't used it before. I know there was an issue submitted to ActiveAdmin regarding this feature. I'm fine with this as is if it meets the coverage requirements. Adding an actual test for the feature can occur separately, it shouldn't delay this. Let me know what you prefer to do here.

tagliala committed:
Test singleton belongs to helper

Use the helper instead of passing the `singleton: true` option for
multiple test cases.

tagliala commented on test/url_helpers_test.rb:
```diff
@@ -106,7 +106,7 @@ class Spot < ModelBase
 class SpotsController < InheritedResources::Base
   belongs_to :house do
-    belongs_to :dishwasher, singleton: true do
+    singleton_belongs_to :dishwasher do
```
@javierjulio after a second look, I've noticed that singleton belongs to were actually tested but only through the `singleton: true` option.

Since `singleton: true` is used in multiple classes and tests, I've changed one to use the helper instead

javierjulio replied:
@tagliala that's awesome! Sounds good to me. Thanks again for seeing this through. Let me know once this is ready for review.

javierjulio reviewed (approved):
Thanks!
